src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -1458,6 +1458,108 @@ void C2_MacroAssembler::string_equals(Register a1, Register a2,
   BLOCK_COMMENT("} string_equals");
 }
 
+void C2_MacroAssembler::arrays_hashcode(Register ary, Register cnt,
+                                        Register result, Register tmp1, Register tmp2,
+                                        Register tmp3, Register tmp4, Register tmp5,
+                                        Register tmp6,
+                                        BasicType eltype)
+{
+  assert_different_registers(ary, cnt, result, tmp1, tmp2, tmp3, tmp4, tmp5, tmp6);
+
+  const int elsize = arrays_hashcode_elsize(eltype);
+
+  switch (eltype) {
+  case T_BOOLEAN: BLOCK_COMMENT("arrays_hashcode(unsigned byte) {"); break;
+  case T_CHAR:    BLOCK_COMMENT("arrays_hashcode(char) {");          break;
+  case T_BYTE:    BLOCK_COMMENT("arrays_hashcode(byte) {");          break;
+  case T_SHORT:   BLOCK_COMMENT("arrays_hashcode(short) {");         break;
+  case T_INT:     BLOCK_COMMENT("arrays_hashcode(int) {");           break;
+  default:        BLOCK_COMMENT("arrays_hashcode {");                break;
+    ShouldNotReachHere();
+  }
+
+  const int stride = 4;
+  const Register pow31_3_4 = tmp6;
+  const Register pow31_2   = tmp5;
+  const Register chunk     = tmp4;
+
+  Label DONE, TAIL, TAIL_LOOP, WIDE_LOOP;
+
+  // result has a value initially
+
+  beqz(cnt, DONE);
+
+  lw(pow31_2, ExternalAddress(StubRoutines::riscv::arrays_hashcode_powers_of_31()
```

Comment:
Now you don't need this `lw` anymore, as 961 will fit in an immediate, so `mv pow31_2, 961` should be fine.

Reply:
Agreed.

Reply:
Done.